plugin.video.carnet-meduza/resources/lib/carnet-meduza.py

```diff
@@ -0,0 +1,264 @@
+	playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+	player = xbmc.Player()
+	playlist.clear()
+	requestChannel = requests.get(api_base_url + 'channel/?id=' + channel_id + '&uid=' + api_key)
+	channel = requestChannel.json()
+	channel_index = channel['index']
+	schedule = channel['raspored']
+	channel_offset = channel['offset']
+	schedule_gen = ((i, j) for i, j in enumerate(schedule))
+	for i, j in schedule_gen:
+	        li = xbmcgui.ListItem(j['naslov'], 'label1', thumbnailImage = j['slika'])
+		itemArgs = {
+                'title': j['naslov'].encode('utf-8'),
+                'plot': j['opis'].encode('utf-8'),
+                'tracknumber': i 
+            	}
+		li.setInfo('video', itemArgs)
+		playlist.add(url=j['stream_url'],listitem=li)
+        player.play(item=playlist,startpos=channel_index)
+	player.seekTime(channel_offset)
+
+base_url = sys.argv[0]
+addon_handle = int(sys.argv[1])
+args = urlparse.parse_qs(sys.argv[2][1:])
+#set as video addon
+xbmcplugin.setContent(addon_handle, 'videos')
+
+addon=xbmcaddon.Addon('plugin.video.carnet-meduza')
+api_base_url = 'https://meduza.carnet.hr/index.php/api/'
+category_image_base_url = 'https://meduza.carnet.hr/uploads/images/categories/'
+# api_key == device_id from register (import) device == uid
+api_key = addon.getSetting('apikey')
+
+mode = args.get('mode', None)
+
+dir_recommends = addon.getLocalizedString(30201)
+dir_categories = addon.getLocalizedString(30202)
+dir_channels = addon.getLocalizedString(30203)
+dir_search = addon.getLocalizedString(30204)
+dir_settings = addon.getLocalizedString(30205)
+
+# get api lang property (en|hr)
+active_lang = xbmc.getLanguage()
+if active_lang == 'Croatian':
+	lang_api_prop = 'naziv'
+else:
+	lang_api_prop = 'naziv_en'
+
+if mode is None:
+    url = build_url({'mode': 'folder', 'foldername': dir_recommends})
+    li = xbmcgui.ListItem(dir_recommends, iconImage='DefaultVideo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': 'folder', 'foldername': dir_categories})
+    li = xbmcgui.ListItem(dir_categories, iconImage='DefaultAddonVideo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': 'folder', 'foldername': dir_channels})
+    li = xbmcgui.ListItem(dir_channels, iconImage='DefaultAddonTvInfo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': dir_search, 'foldername': dir_search})
+    li = xbmcgui.ListItem(dir_search, iconImage='DefaultAddonsSearch.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+	
+    url = build_url({'mode': dir_settings, 'foldername': dir_settings})
+    li = xbmcgui.ListItem(dir_settings, iconImage='DefaultAddonProgram.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li)
+	
```

Comment:
Just a comment here, not really an issue. You may find it easier to build a list of directory items use addDirectoryItems to add them.  Though your list is short, addDirectoryItems is much more efficient and the resulting code is cleaner. Please see http://mirrors.kodi.tv/docs/python-docs/16.x-jarvis/xbmcplugin.html#-addDirectoryItems for info.